src/codeu/chat/client/commandline/Chat.java

```diff
@@ -40,6 +43,44 @@ public Chat(Context context) {
 		this.panels.push(createRootPanel(context));
 	}
 
+	// RECREATE 
+	//
+	// Recreates the server's last known state to display for the client 
+	// upon relaunching the client 
+	// Will not display anything if there was no information saved 
+	//
+	//
+	public void recreate(Context context) {
+		Hashtable<Uuid, String> allUsers = new Hashtable<Uuid, String>();
```

Comment:
What is this used for?

Also, you should generally use a `HashMap` instead of a `Hashtable`. `Hashtable` is older and generally considered legacy code. More info here: https://stackoverflow.com/questions/40471/differences-between-hashmap-and-hashtable